clang/lib/CConv/ArrayBoundsInferenceConsumer.cpp

```diff
@@ -536,16 +536,17 @@ void LocalVarABVisitor::addUsedParmVarDecl(Expr *CE) {
       NonLengthParameters.insert(PVD);
 }
 
-bool LocalVarABVisitor::VisitIfStmt(IfStmt *IFS) {
-  if (BinaryOperator *BO = dyn_cast<BinaryOperator>(IFS->getCond())) {
```

Comment:
To be clear: You are removing mention of an if/conditional because the cases of interest will be covered by `VisitBinaryOperator` anyway, e.g., if the operator is used in a conditional guard?

Reply:
Yes!